src/System.Binary.Base64/System/Binary/BufferExtensions.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Buffers;
+using System.Collections.Sequences;
+
+// This should most likely be in System.Text.Primitives and not here. Will move once Base64 moves.
+namespace System.Binary.Base64
+{
+    public static class BufferExtensions
+    {
+        const int stackLength = 32;
+
+        public static void Pipe(this ITransformation transformation, ReadOnlyBytes source, IOutput destination)
+        {
+            int afterMergeSlice = 0;
+            ReadOnlySpan<byte> remainder;
+            Span<byte> stackSpan;
+
+            unsafe
+            {
+                byte* stackBytes = stackalloc byte[stackLength];
```

Comment:
Do we have to do this here? What if the stack buffer does not endup being needed?